### ISSUE TITLE
Juddering UI due to scroll bar

### ISSUE THREAD

**Issue by deosrc** [blakeblackshear/frigate#1657]
**Describe the bug**

The UI is constantly adding and removing the scroll bar which is causing a juddering:

![ezgif-3-5caf3e5970de](https://user-images.githubusercontent.com/50599779/131378388-d98d1e20-ce99-4079-b1ea-68bce548cb8b.gif)

(This is actually happening much faster but the frame rate of the GIF reduced the effect)

**Version of frigate**

0.9.0-46fe06e

**Config file**
Include your full config file wrapped in triple back ticks.
```yaml
    mqtt:
      host: mqtt
      user: frigate
      password: '{FRIGATE_MQTT_PASSWORD}'
    hwaccel_args:
      - -c:v
      - h264_v4l2m2m
    detectors:
      cpu1:
        type: cpu
      cpu2:
        type: cpu
      cpu3:
        type: cpu
    objects:
      track:
        - person
        - vehicle
        - animal
    record:
      enabled: true
      retain_days: 0
      events:
        enabled: true
        objects:
          - person
          - vehicle
          - animal
        retain:
          animal: 3
    birdseye:
      enabled: false
    cameras:
      front:
        ffmpeg:
          inputs:
            - path: 'rtsp://frigate:{FRIGATE_CCTV_PASSWORD}@cctv/Streaming/Channels/101'
              roles:
                - rtmp
                - detect
                - record
        detect:
          width: 768
          height: 432
          fps: 2
        motion:
          mask:
            - 14,19,14,42,218,42,218,19 # Timestamp
      side:
        ffmpeg:
          inputs:
            - path: 'rtsp://frigate:{FRIGATE_CCTV_PASSWORD}@cctv/Streaming/Channels/201'
              roles:
                - rtmp
                - detect
                - record
        detect:
          width: 768
          height: 432
          fps: 2
        motion:
          mask:
            - 14,19,14,42,218,42,218,19 # Timestamp
      garden_1:
        ffmpeg:
          inputs:
            - path: 'rtsp://frigate:{FRIGATE_CCTV_PASSWORD}@cctv/Streaming/Channels/301'
              roles:
                - rtmp
                - detect
                - record
        detect:
          width: 768
          height: 432
          fps: 2
        motion:
          mask:
            - 14,19,14,42,218,42,218,19 # Timestamp
        objects:
          track:
            - person
            - animal
      garden_2:
        ffmpeg:
          inputs:
            - path: 'rtsp://frigate:{FRIGATE_CCTV_PASSWORD}@cctv/Streaming/Channels/401'
              roles:
                - rtmp
                - detect
                - record
        detect:
          width: 768
          height: 432
          fps: 2
        motion:
          mask:
            - 14,19,14,42,218,42,218,19 # Timestamp
      shed:
        ffmpeg:
          inputs:
            - path: 'rtsp://frigate:{FRIGATE_CCTV_PASSWORD}@cctv/Streaming/Channels/501'
              roles:
                - rtmp
                - detect
                - record
        detect:
          width: 768
          height: 432
          fps: 2
        motion:
          mask:
            - 14,19,14,42,218,42,218,19 # Timestamp
```

**Frigate container logs**
```
Include relevant log output here
```

**Frigate stats**
```json
Output from frigate's /api/stats endpoint
```

**FFprobe from your camera**

Run the following command and paste output below
```
ffprobe <stream_url>
```

**Screenshots**

**Computer Hardware**
 - OS: [e.g. Ubuntu, Windows] Pop!_OS
 - Install method: [e.g. Addon, Docker Compose, Docker Command] Kubernetes
 - Virtualization: [e.g. Proxmox, Virtualbox] N/A
 - Coral Version: [e.g. USB, PCIe, None] None
 - Network Setup: [e.g. Wired, WiFi] Wired

**Camera Info:**
 - Manufacturer: [e.g. Dahua] Hikvision DVR
 - Model: [e.g. IPC-HDW5231R-ZE]
 - Resolution: [e.g. 720p] 720p
 - FPS: [e.g. 5]

**Additional context**

Information from browser developer tools:

`html` tag computed size is flicking between 1920x883 and 1908x886
`width` and `height` attributes on `<canvas data-testid="comeraimage-canvas">` seem to be flicking between 528x297 and 532x299 for all cameras (same models so expected that the values would match).

**Comment by Sinamics** [comment]
Can you confirm of this flickering stops if you zoom in or out in the browser?

**Comment by deosrc** [comment]
> Can you confirm of this flickering stops if you zoom in or out in the browser?

Yes that's correct. It is only present at 100% zoom level. It also disappears if I manually set a width on the HTML tag, even if that value is one of those above (1920 or 1908).

My best guess based on the way it is reacting is that it is something related to how it is sizing the canvas once it has the first frame of the video feed(s?). This causes it to expand off the bottom of the page and need the scroll bar. When the scroll bar appears, it then resizes to maintain the aspect ratio and fit 3 feeds side by side which then reduces the size of the elements so that it doesn't need the scroll bar any more so removes it. Then repeat...

Just happens to be that my maximised browser dimensions and video aspect ratios perfectly align to cause this.

**Comment by ozett** [comment]
my jiggering went away after resizing the browserwindow
ver DEBUG 0.9.0-46FE06E

![fritagetremble2](https://user-images.githubusercontent.com/5937811/131391750-b08e6614-775b-4b8d-9e87-b5afe0f256fe.gif)

**Comment by Sinamics** [comment]
@deosrc @ozett 
Im able to reproduce this behavior. 
I belive this is related to the calculated `availableWidth` from the ResizeObserver. 
https://github.com/blakeblackshear/frigate/blob/bddde74c06806552188772f8c3cc1d1bfa583ff1/web/src/components/CameraImage.jsx#L13

I will do some more testing the next days and hopefully provide an PR for this.

**Comment by deosrc** [comment]
Didn't realise this had gone stale. Still an issue in 0.9.4.

**Comment by alexelite** [comment]
Same issue here. 0.9.4-26AE608

**Comment by scaytrase** [comment]
Still an issue here :(  0.11.1 (stable)

**Comment by mkaatman** [comment]
It needs scrollbar-gutter stable applied.

https://css-tricks.com/almanac/properties/s/scrollbar-gutter/

**Comment by NickM-27** [comment]
Fixed for 0.12

**Comment by mscott5861** [comment]
Seeing this Issue in 0.11.1-2EADA21

**Comment by NickM-27** [comment]
> Seeing this Issue in 0.11.1-2EADA21

Right, it is fixed in 0.12
